src/modding/ModLoader.cs

```diff
@@ -0,0 +1,278 @@
+    }
+
+    private void UnLoadMod(string name)
+    {
+        var info = LoadModInfo(name);
+
+        if (info == null)
+        {
+            GD.PrintErr("Can't load mod due to failed info reading: ", name);
+            return;
+        }
+
+        if (loadedModAssemblies.TryGetValue(name, out var mod))
+        {
+            GD.Print("Unloaded mod contained an assembly, sending it the unload method call");
+
+            try
+            {
+                if (!mod.Unload())
+                {
+                    GD.PrintErr("Mod's (", name, ") assembly unload method call failed");
+                }
+            }
+            catch (Exception e)
+            {
+                GD.PrintErr("Mod's (", name, ") assembly unload method call failed with an exception: ", e);
+            }
+
+            loadedModAssemblies.Remove(name);
+        }
+    }
+
+    private void RunCodeModFirstRunCallbacks(IMod mod)
+    {
+        var scene = modInterface.CurrentScene;
+
+        mod.CanAttachNodes(scene);
+    }
+
+    private bool CreateModInstance(string name, FullModDetails info, Assembly assembly)
+    {
+        var className = info.Info.AssemblyModClass;
+
+        var type = assembly.GetTypes().FirstOrDefault(t => t.Name == className);
+
+        if (type == null)
+        {
+            GD.Print("No class with name \"", className, "\" found, can't finish loading mod assembly");
+            return false;
+        }
+
+        try
+        {
+            var mod = (IMod)Activator.CreateInstance(type);
+
+            if (!mod.Initialize(modInterface, info.Info))
+            {
+                GD.PrintErr("Mod's (", name, ") initialize method call failed");
+            }
+
+            loadedModAssemblies.Add(name, mod);
+
+            if (!firstExecute)
+            {
+                // We need to do the actions that would normally be delayed if this mod was loaded after game startup
+                RunCodeModFirstRunCallbacks(mod);
+            }
+        }
+        catch (Exception e)
+        {
+            GD.PrintErr("Mod's (", name, ") initialization failed with an exception: ", e);
+        }
+
+        return true;
+    }
+
+    private void LoadPckFile(string path)
+    {
+        GD.Print("Loading mod .pck file: ", path);
+
+        if (!ProjectSettings.LoadResourcePack(path))
+        {
+            GD.PrintErr(".pck loading failed");
+        }
+    }
+
+    private Assembly LoadCodeAssembly(string path)
+    {
+        path = ProjectSettings.GlobalizePath(path);
+
+        if (!File.Exists(path))
+        {
+            GD.PrintErr("Can't load assembly from non-existent path: ", path);
+            throw new ArgumentException("Invalid given assembly path");
+        }
+
+        GD.Print("Loading mod C# assembly from: ", path);
+
+        // This version doesn't load the classes into the current assembly find path (or whatever it is properly called
+        // in C#)
+        var result = Assembly.LoadFile(path);
```

Comment:
I think we want to use `LoadFrom` here. The difference is very confusing and the documentation doesn't help at all.

Apparently (https://stackoverflow.com/a/15093022) `LoadFrom` also loads the required dependencies (e.g. inter-mod dependencies)

Reply:
Changed.